src/ContainerAppHelper.ts

```diff
@@ -0,0 +1,434 @@
+    public async createRunnableAppImage(
+        imageToDeploy: string,
+        appSourcePath: string,
+        runtimeStack: string) {
+        toolHelper.writeDebug(`Attempting to create a runnable application image using the Oryx++ Builder with image name "${imageToDeploy}"`);
+        try {
+            let telemetryArg = toolHelper.getTelemetryArg();
+            if (this.disableTelemetry) {
+                telemetryArg = `ORYX_DISABLE_TELEMETRY=true`;
+            }
+            let command = `build ${imageToDeploy} --path ${appSourcePath} --builder ${ORYX_BUILDER_IMAGE} --run-image mcr.microsoft.com/oryx/${runtimeStack} --env ${telemetryArg}`;
+            await util.executeAndThrowIfError(`${PACK_CMD} ${command}`);
+        } catch (err) {
+            toolHelper.writeError(err.message);
+            throw err;
+        }
+    }
+
+    /**
+     * Using a Dockerfile that was provided or found at the root of the application source,
+     * creates a runable application image.
+     * @param imageToDeploy - the name of the runnable application image that is created and can be later deployed
+     * @param appSourcePath - the path to the application source on the machine
+     * @param dockerfilePath - the path to the Dockerfile to build and tag with the provided image name
+     */
+    public async createRunnableAppImageFromDockerfile(
+        imageToDeploy: string,
+        appSourcePath: string,
+        dockerfilePath: string) {
+        toolHelper.writeDebug(`Attempting to create a runnable application image from the provided/found Dockerfile "${dockerfilePath}" with image name "${imageToDeploy}"`);
+        try {
+            let command = `docker build --file ${dockerfilePath} ${appSourcePath} --tag ${imageToDeploy}`;
+            await util.executeAndThrowIfError(command);
+            toolHelper.writeDebug(`Successfully created runnable application image from the provided/found Dockerfile "${dockerfilePath}" with image name "${imageToDeploy}"`);
+        } catch (err) {
+            toolHelper.writeError(err.message);
+            throw err;
+        }
+    }
+
+    /**
+     * Determines the runtime stack to use for the runnable application image.
+     * @param appSourcePath - the path to the application source on the machine
+     * @returns a string representing the runtime stack that can be used for the Oryx MCR runtime images
+     */
+    public async determineRuntimeStackAsync(appSourcePath: string): Promise<string> {
+        toolHelper.writeDebug('Attempting to determine the runtime stack needed for the provided application source');
+        try {
+            let dockerTool: string = await toolHelper.which("docker", true);
+            // Use 'oryx dockerfile' command to determine the runtime stack to use and write it to a temp file
+            let command = `run --rm -v ${appSourcePath}:/app ${ORYX_CLI_IMAGE} /bin/bash -c \"oryx dockerfile /app | head -n 1 | sed 's/ARG RUNTIME=//' >> /app/oryx-runtime.txt\"`
```

Comment:
nit: same comment here about escaping the double quotes

Reply:
Removed double quotes.